api/envoy/config/filter/http/ext_authz/v2alpha/ext_authz.proto

```diff
@@ -88,3 +90,23 @@ message HttpService {
   // authorization server. Note that these will override the headers coming from the downstream.
   repeated envoy.api.v2.core.HeaderValue authorization_headers_to_add = 6;
 }
+
+// Extra settings for the check request on a per virtualhost/route/weighter-cluster level.
+message ExtAuthPerRoute {
+  oneof override {
+    option (validate.required) = true;
+
+    // Disable the ext auth filter for this particular vhost or route.
+    bool disabled = 1 [(validate.rules).bool.const = true];
+
+    // Check request settings for this route.
+    CheckSettings check_settings = 2 [(validate.rules).message.required = true];
+  }
+}
+
+// Extra settings for the check request
+message CheckSettings {
+  // Context extensions to set on the CheckRequest's
+  // :ref:`AttributeContext<envoy_api_msg_service.auth.v2alpha.AttributeContext>`
+  map<string, string> context_extensions = 1;
```

Comment:
As I understand this will only be effective when the filter is used with the gRPC client. If that is correct, could you please add a comment explaining the behaviour?

Reply:
Awesome, thanks!